src/v2/view-builder/views/DeviceChallengePollView.js

```diff
@@ -48,36 +54,54 @@ const Body = BaseForm.extend(Object.assign(
     },
 
     doChallenge () {
-      const deviceChallenge = this.deviceChallengePollRemediation.relatesTo.value;
-      switch (deviceChallenge.challengeMethod) {
+      switch (this.deviceChallenge.challengeMethod) {
       case 'LOOPBACK':
         this.title = loc('signin.with.fastpass', 'login');
         this.add('<div class="spinner"></div>');
-        this.doLoopback(deviceChallenge.domain, deviceChallenge.ports, deviceChallenge.challengeRequest);
+        this.doLoopback(this.deviceChallenge.domain, this.deviceChallenge.ports, this.deviceChallenge.challengeRequest);
         break;
       case 'CUSTOM_URI':
         this.title = loc('customUri.title', 'login');
         this.subtitle = loc('customUri.subtitle', 'login');
         this.add(`
           {{{i18n code="customUri.content" bundle="login"}}}
         `);
-        this.customURI = deviceChallenge.href;
+        this.customURI = this.deviceChallenge.href;
         this.doCustomURI();
         break;
       case 'UNIVERSAL_LINK':
-        this.title = loc('universalLink.title', 'login');
-        this.add(`
-          {{{i18n code="universalLink.content" bundle="login"}}}
-        `);
-        this.add(createButton({
-          className: 'ul-button button button-wide button-primary',
-          title: loc('universalLink.button', 'login'),
-          click () {
-            // only window.location.href can open universal link in iOS/MacOS
-            // other methods won't do, ex, AJAX get or form get (Util.redirectWithFormGet)
-            Util.redirect(deviceChallenge.href);
-          }
-        }));
+        if (this.currentRemediationName === RemediationForms.USER_VERIFICATION_CHALLENGE_POLL) {
```

Comment:
In general, I dont remember share view between different ion form if you have to compare form in the view.
so why not create two views instead ?